src/posix/platform/include/openthread/openthread-system.h

```diff
@@ -81,27 +81,44 @@ typedef struct otPlatformConfig
 } otPlatformConfig;
 
 /**
- * This function performs all platform-specific initialization of OpenThread's drivers.
+ * This function performs all platform-specific initialization of OpenThread's drivers and initialize the OpenThread
+ * instance.
  *
  * @note This function is not called by the OpenThread library. Instead, the system/RTOS should call this function
  *       when initialization of OpenThread's drivers is most appropriate.
  *
  * @param[in]  aPlatformConfig  Platform configuration structure.
  *
- * @returns A pointer to the OpenThread instance.
+ * @sa otSysGetInstance
  *
  */
-otInstance *otSysInit(otPlatformConfig *aPlatformConfig);
+void otSysInit(otPlatformConfig *aPlatformConfig);
 
 /**
- * This function performs all platform-specific deinitialization for OpenThread's drivers.
+ * This function performs all platform-specific deinitialization for OpenThread's drivers and finalizes the OpenThread
+ * instance.
```

Comment:
nit: The OT instance is finalized before the platform deinitialization is done. Maybe we can swap the steps in the comment: 
```
This function finalizes the OpenThread instance and performs all platform-specific deinitialization for OpenThread's drivers.
```